### ISSUE TITLE
Unable to build example on macos 10.14.5

### ISSUE THREAD

**Issue by brianbruggeman** [indygreg/PyOxidizer#11]
I'm pretty new to rust, and I lack the experience with rust to debug this.  I assume its something simple; I thought the following should generally "just work".

```
$ cargo install pyoxidizer
$ pyoxidizer init hello
$ cd hello
$ pyoxidizer build
    ... stuff ...
    Updating crates.io index
    Updating git repository `https://github.com/indygreg/PyOxidizer.git`
error: failed to load source for a dependency on `cpython`

Caused by:
  Unable to update https://github.com/indygreg/PyOxidizer.git?rev=

Caused by:
  failed to parse revision specifier - Invalid pattern ''; class=Invalid (3); code=InvalidSpec (-12)
error: cargo build failed
```

$ rustc --version
rustc 1.37.0-nightly (de7c4e423 2019-06-23)

$ cargo --version
cargo 1.37.0-nightly (807429e1b 2019-06-11)

$ pyoxidizer --version
PyOxidizer 0.1 (commit )

$ python --version
Python 3.7.3

**Comment by indygreg** [comment]
I think I know what the problem is. I'm pretty confident that if you install from a Git clone or via `cargo install --git https://github.com/indygreg/PyOxidizer.git --branch master pyoxidizer` it should work. I'll try to have a fix published ~8 hours from now.

**Comment by snorrwe** [comment]
I have the same issue on Windows, tried installing pyoxidizer via the command you suggested, but alas no joy.

**Comment by brianbruggeman** [comment]
@snorrwe on the mac, I was successful after running (note the additional `--force` option):
```
cargo install --force --git https://github.com/indygreg/PyOxidizer.git --branch master pyoxidizer
```
Note that this is just a vanilla "hello world" / REPL project with no changes.

I also tried wrapping a `click` based python package and I was not successful; it was missing the module.  So I'll need to dig a bit to figure out why there.

**Comment by indygreg** [comment]
So the issue here is that generated projects need to refer to the PyOxidizer Git repository so they can pull in the forked version of the `rust-cpython` crate that is vendored into the repository. When `pyoxidizer` is itself built, we attempt to resolve the Git commit it corresponds to. That is somehow failing, resolving the commit to an empty string (``), and things blow up. I believe this only happens in builds from crates, as builds from checked out Git repositories should _just work_. But alas, the mechanism used to resolve the Git commit uses `git`. So if `git` is not available in the current shell, the resolution will fail and it will use `UNKNOWN` for the commit value. I have my work cut out for me...

**Comment by indygreg** [comment]
I just released PyOxidizer 0.1.1 to crates.io. `cargo install --force pyoxidizer` should work now. Please file a new issue if it doesn't.

If you do run into issues where the build complains about a missing reference to `https://github.com/indygreg/PyOxidizer.git`, you can edit the `pyembed/Cargo.toml` and update the references to this repository to match something that exists. `Cargo.toml` Git dependencies accept either `commit`, `tag`, or `branch` fields to define the Git commit to use (https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories).
